### ISSUE TITLE
Schema incorrectly generates `null` instead of a number

### ISSUE THREAD

**Issue by jgarplind** [soc221b/zod-schema-faker#281]
**Describe the bug** A clear and concise description of what the bug is.

This schema (reduced for reproduciton purposes):
```
z.object({
  id: z.string(),
  assignees: z.array(
    z.object({
      id: z.number(),
    }),
  ),
});
```

always generates `null` instead of a proper number for the inner `id` property.

**To Reproduce** Steps to reproduce the behavior:

1. `const theGeneratedFake = fake(theSchemaAbove)`
2. `conole.log(theGeneratedFake)` --> `// {"id":"volup","assignees":[{"id":null},{"id":null}]}`
3. `theSchemaAbove.parse(theGeneratedFake)` --> ZodError

**Expected behavior** A clear and concise description of what you expected to happen.

**Screenshots** If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context** Add any other context about the problem here.
I also tried putting `min(1)` on the `assignees` array to cause the problem to reproduce consistenly, but then I got `Array must contain at least 1 element(s)` errors instead, implying that that property is also not properly respected.

Sorry for being brief.

**Comment by soc221b** [comment]
Since `Infinity` and some other edge case values are considered valid for `z.number()`, and `JSON.stringify(Infinity)` produces `'null'`, could you please check if it is being stringified?

(If you want to produce finite numbers: `z.number().finite()`)

**Comment by jgarplind** [comment]
Right, I did `console.log(JSON.stringify(fake))` to be able to see the nested properties, so I think there is some merit to your assumption.

There should however be no stringification going on in my app logic under test, but maybe I'm mistaken.

I opted to try a similar library which worked for my use case out of the box, so I don't plan to spend more time on this right now, sorry. Just wanted to share my experience for your awareness.
